gammapy/cube/models.py

```diff
@@ -375,13 +378,25 @@ def apply_psf(self, npred):
         """Convolve npred cube with PSF"""
         return self.psf.apply(npred)
 
+    def apply_edisp(self, npred):
+        """Convolve npred cube with edisp"""
+        return np.dot(npred.data, self.edisp.pdf_matrix)
```

Comment:
@AtreyeeS - From the description of [numpy.dot](https://docs.scipy.org/doc/numpy/reference/generated/numpy.dot.html), I think you might have to move the energy axis on the map to the end. I didn't try, but something like this should do it:
```
a = np.rollaxis(npred.data, 0, 4)
```
Note that the goal is only to work on 3D maps for now.

For the edisp, I'm not sure if you need
```
b = self.edisp.pdf_matrix
```
with or without transpose.
I guess an example where you have an off-diagonal non-zero entry will show this.

Let me know if you get stuck here, and I'll have a closer look.